packages/react-component-library/src/components/CheckboxRadioBase/CheckboxRadioBase.tsx

```diff
@@ -124,6 +131,7 @@ export const CheckboxRadioBase = React.forwardRef<
                 onBlur={onBlur}
                 disabled={isDisabled}
                 data-testid={`${type}-input`}
+                checked={defaultChecked ? undefined : isChecked}
```

Comment:
`defaultChecked` and `checked` are mutually exclusive.

Reply:
Would be good to explicitly test against `undefined` I think (as earlier comment – I think `false` should be respected):

```suggestion
                checked={defaultChecked !== undefined ? undefined : isChecked}
```

(It might even better to define e.g. an `const isControlled = ...` above and use that in both places.)